### ISSUE TITLE
Make set_user_agent return the new user agent

### ISSUE THREAD

**Issue by lucaswerkmeister** [legoktm/toolforge#7]
Some libraries, such as `mwoauth` or `mwapi`, choose their own default user agent if you don’t specify one separately, instead of relying on the underlying `requests` API’s default user agent. To avoid this, in several tools I’ve needed to do the following:

```python
toolforge.set_user_agent('my-tool', email='mail@lucaswerkmeister.de')                                                                                                             
user_agent = requests.utils.default_user_agent()

# ...

... = mwoauth.initiate(..., user_agent=user_agent)
```

This would be slightly prettier, I think, if `set_user_agent` returned the user agent directly, so I didn’t have to extract it out of the `requests` library.

**Comment by legoktm** [comment]
Good idea, implemented, and included in the 4.2.0 release.

**Comment by lucaswerkmeister** [comment]
Awesome, thank you :)
